### PR TITLE
Add busy_timeout accelerator param for Sqlite

### PR DESCRIPTION
**Pull request by Sevenannn**
## 🗣 Description

TODO
- [x] Update the patch of datafusion table provider after this is merged: https://github.com/datafusion-contrib/datafusion-table-providers/pull/121
- [ ] Doc update: https://github.com/spiceai/docs/pull/426

* Add `busy_timeout` acceleration parameter for Sqlite, this will solve the concurrent writing failure caused by file level lock when writing into the same Sqlite file
* Temporarily remove the mysql enum types to unblock e2e test ci. The datafusion table provider patch update includes update of using arrow dictionary type for mysql enum, it fails on duckdb and sqlite because they don't have arrow dictionary type write support. Bugs created for duckdb and sqlite
  - https://github.com/spiceai/spiceai/issues/2891
  - https://github.com/spiceai/spiceai/issues/2889

## 🔨 Related Issues

- https://github.com/spiceai/spiceai/issues/2761

### CODE REVIEW

**Review by peasee — DISMISSED** [review]

File: crates/runtime/src/dataaccelerator/sqlite.rs [hunk 1]
```diff
@@ -66,6 +66,9 @@ pub enum Error {
     #[snafu(display("The \"sqlite_file\" acceleration parameter is a directory."))]
     InvalidFileIsDirectory,
 
+    #[snafu(display("The \"busy_timeout\" acceleration parameter is not a valid duration."))]
```
Comment (lukekim) [inline crates/runtime/src/dataaccelerator/sqlite.rs]:
```suggestion
    #[snafu(display("The \"busy_timeout\" acceleration parameter value is not a valid duration."))]
```

File: crates/runtime/src/dataaccelerator/sqlite.rs [hunk 2]
```diff
@@ -66,6 +66,9 @@ pub enum Error {
     #[snafu(display("The \"sqlite_file\" acceleration parameter is a directory."))]
```
Comment (lukekim) [inline crates/runtime/src/dataaccelerator/sqlite.rs]:
```suggestion
    #[snafu(display("The \"sqlite_file\" acceleration parameter value is a directory."))]
```

File: test/scripts/setup-data-mysql.sql [hunk 3]
```diff
@@ -1,3 +1,4 @@
+-- TODO: Add MySQL enum type back when DucKDB & SQLite accelerator support accelerating arrow dictionary type
```
Comment (lukekim) [inline test/scripts/setup-data-mysql.sql:1]: Link to issue?

Comment (Sevenannn) [reply test/scripts/setup-data-mysql.sql:1]: Linked

**Review by peasee — APPROVED** [review]

**Review by phillipleblanc — APPROVED** [review]

**Pull request merged**